Overwrite output lines for successful tests

kwiberg opened the pull request:
This gives a tiny speed boost, since there's less scrolling, but more
importantly it means that interesting error messages from failing
tests won't scroll off the screen just because they're followed by
hundreds of tests that don't fail.

pbos commented:
lgtm, squash

pbos commented on gtest-parallel:
```diff
@@ -23,8 +23,48 @@ import threading
+# width, and will overwrite an immediately preceding transient line;
+# however, if a transient line is followed by a permanent line, or a
+# permanent by a transient, nothing will be overwritten. Finally, if
+# the line width can't be determined, lines are never overwritten.
```
Can you make this comment fit into like half of the lines? Also make sure you say why you're doing it instead of explaining the code. What does this result in?

kwiberg commented:
Is this better?

pbos commented on gtest-parallel:
```diff
@@ -23,8 +23,47 @@ import threading
+    return None
+
+# Output transient and permanent lines of text to an output file
+# object. If several transient lines are written in sequence, the new
```
remove "to an output file object"

pbos commented on gtest-parallel:
```diff
@@ -23,8 +23,47 @@ import threading
+# (tests failing).
+class Outputter(object):
+  def __init__(self, out):
+    self.__out = out  # The output file object.
```
rename this __out_file and remove the comment

pbos commented on gtest-parallel:
```diff
@@ -23,8 +23,47 @@ import threading
+class Outputter(object):
+  def __init__(self, out):
+    self.__out = out  # The output file object.
+    self.__transient = False  # The last line written was a transient.
```
__last_line_was_transient, remove comment

pbos commented on gtest-parallel:
```diff
@@ -23,8 +23,47 @@ import threading
+  def __init__(self, out):
+    self.__out = out  # The output file object.
+    self.__transient = False  # The last line written was a transient.
+    self.__width = term_width(out)  # Line width, or None.
```
or None if not a tty.

pbos commented:
lgtm with last comments addressed, please rebase/squash

kwiberg committed:
Overwrite output lines for successful tests

This gives a tiny speed boost, since there's less scrolling, but more
importantly it means that interesting error messages from failing
tests won't scroll off the screen just because they're followed by
hundreds of tests that don't fail.
